yt_dlp/cookies.py

```diff
@@ -740,7 +742,9 @@ def _get_linux_desktop_environment(env, logger):
     xdg_current_desktop = env.get('XDG_CURRENT_DESKTOP', None)
     desktop_session = env.get('DESKTOP_SESSION', None)
     if xdg_current_desktop is not None:
-        xdg_current_desktop = xdg_current_desktop.split(':')[0].strip()
+        xdg_current_desktop = [env.strip() for env in xdg_current_desktop.split(':')]
+        xdg_current_desktop = next(
+            (env for env in xdg_current_desktop if env in SUPPORTED_DESKTOP_ENVS), xdg_current_desktop[0])
```

Comment:
Instead of doing this, I think this would be better suited to be doing a `for`, like:
```py
        for part in map(str.strip, xdg_current_desktop.split(':')):
            if part == 'Unity':
                ...
            elif part == 'Deepin':
                return _LinuxDesktopEnvironment.DEEPIN
        else:
            logger.info(f'XDG_CURRENT_DESKTOP is set to an unknown value: "{xdg_current_desktop}"')
```
that way we dont have to duplicate the values and have a single point where we have all cases